### ISSUE TITLE
btVector3::normalize(): Assertion `!fuzzyZero()' failed

### ISSUE THREAD

**Issue by stephengold** [stephengold/Minie#3]
With a jumping character with non-default gravity and a static rigid body with a mesh shape, I get a `btAssert()` failure when the peak of the jump is inside the body's bounding box.  (Only with a debug native library, of course.)

First seen in `TestQ3` and `HelloCollision`, I've boiled it down to the following test case:
```
import com.jme3.app.SimpleApplication;
import com.jme3.bullet.BulletAppState;
import com.jme3.bullet.collision.shapes.CapsuleCollisionShape;
import com.jme3.bullet.collision.shapes.CollisionShape;
import com.jme3.bullet.collision.shapes.MeshCollisionShape;
import com.jme3.bullet.objects.PhysicsCharacter;
import com.jme3.bullet.objects.PhysicsRigidBody;
import com.jme3.input.KeyInput;
import com.jme3.input.controls.ActionListener;
import com.jme3.input.controls.KeyTrigger;
import com.jme3.math.Quaternion;
import com.jme3.math.Vector3f;
import com.jme3.scene.Mesh;
import com.jme3.scene.VertexBuffer;
import java.nio.FloatBuffer;
import jme3utilities.math.MyBuffer;
import jme3utilities.mesh.RectangleMesh;
import jme3utilities.minie.PhysicsDumper;

public class TestIssueXXX extends SimpleApplication implements ActionListener {

    private PhysicsCharacter character;

    public static void main(String[] args) {
        TestIssueXXX app = new TestIssueXXX();
        app.start();
    }

    @Override
    public void simpleInitApp() {
        inputManager.addMapping("Jump", new KeyTrigger(KeyInput.KEY_SPACE));
        inputManager.addListener(this, "Jump");

        BulletAppState bulletAppState = new BulletAppState();
        stateManager.attach(bulletAppState);
        bulletAppState.setDebugEnabled(true);

        Mesh mesh = new RectangleMesh(-999f, 999f, -999f, 999f, 1f);
        FloatBuffer buffer = mesh.getFloatBuffer(VertexBuffer.Type.Position);
        Quaternion rot = new Quaternion().fromAngles(1.5f, 0f, 0f);
        MyBuffer.rotate(buffer, 0, buffer.limit(), rot);
        CollisionShape meshShape = new MeshCollisionShape(mesh);
        PhysicsRigidBody staticBody = new PhysicsRigidBody(meshShape, 0f);
        bulletAppState.getPhysicsSpace().add(staticBody);

        CapsuleCollisionShape capsuleShape = new CapsuleCollisionShape(1f, 3f);
        character = new PhysicsCharacter(capsuleShape, 0.05f);
        character.setGravity(30f);
        character.setPhysicsLocation(new Vector3f(0f, 10f, 0f));
        bulletAppState.getPhysicsSpace().add(character);

        new PhysicsDumper().dump(bulletAppState);
    }

    @Override
    public void onAction(String binding, boolean value, float tpf) {
        if (binding.equals("Jump")) {
            character.jump();
        }
    }
}
```

**Comment by stephengold** [comment]
1. If I change the rotation angle to `FastMath.HALF_PI`, the issue goes away.
2. If I comment out `setGravity()`, the issue goes away.

Here's the stack trace:
```
#0  in __GI_raise() at ../sysdeps/unix/sysv/linux/raise.c:51
#1  in __GI_abort() at abort.c:79
#2  in __assert_fail_base() at assert.c:92
#3  in __GI___assert_fail() at assert.c:101
#4  in btVector3::normalize() at LinearMath/btVector3.h:305
#5  in btQuantizedBvh::walkStacklessQuantizedTreeAgainstRay(btNodeOverlapCallback*, btVector3 const&, btVector3 const&, btVector3 const&, btVector3 const&, int, int) const
    at BulletCollision/BroadphaseCollision/btQuantizedBvh.cpp:557
#6  in btQuantizedBvh::reportBoxCastOverlappingNodex(btNodeOverlapCallback*, btVector3 const&, btVector3 const&, btVector3 const&, btVector3 const&) const
    at BulletCollision/BroadphaseCollision/btQuantizedBvh.cpp:758
#7  in btBvhTriangleMeshShape::performConvexcast(btTriangleCallback*, btVector3 const&, btVector3 const&, btVector3 const&, btVector3 const&)
    at BulletCollision/CollisionShapes/btBvhTriangleMeshShape.cpp:225
#8  in btCollisionWorld::objectQuerySingleInternal(btConvexShape const*, btTransform const&, btTransform const&, btCollisionObjectWrapper const*, btCollisionWorld::ConvexResultCallback&, float)
    at BulletCollision/CollisionDispatch/btCollisionWorld.cpp:712
#9  in btCollisionWorld::objectQuerySingle(btConvexShape const*, btTransform const&, btTransform const&, btCollisionObject*, btCollisionShape const*, btTransform const&, btCollisionWorld::ConvexResultCallback&, float)
    at BulletCollision/CollisionDispatch/btCollisionWorld.cpp:609
#10 in btGhostObject::convexSweepTest(btConvexShape const*, btTransform const&, btTransform const&, btCollisionWorld::ConvexResultCallback&, float) const
    at BulletCollision/CollisionDispatch/btGhostObject.cpp:132
#11 in btKinematicCharacterController::stepUp(btCollisionWorld*)
    at BulletDynamics/Character/btKinematicCharacterController.cpp:286
#12 in btKinematicCharacterController::playerStep(btCollisionWorld*, float)
    at BulletDynamics/Character/btKinematicCharacterController.cpp:777
#13 in btKinematicCharacterController::updateAction(btCollisionWorld*, float)
    at BulletDynamics/Character/btKinematicCharacterController.h:124
#14 in btDiscreteDynamicsWorld::updateActions(float)
    at BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp:590
#15 in btDiscreteDynamicsWorld::internalSingleStepSimulation(float)
    at BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp:489
```

**Comment by stephengold** [comment]
`setGravity(29.9f)` works, `setGravity(30.1f)` works, but `setGravity(30f)` doesn't!

**Comment by stephengold** [comment]
To fail sooner, use `setGravity(120f)`.

**Comment by stephengold** [comment]
And `setPhysicsLocation(new Vector3f(0f, 0.8f, 0f))`

**Comment by stephengold** [comment]
During the jump, the Y component of the linear velocity starts at 10. With `setGravity(120f)` it decreases by 0.5 on every physics tick. The crash probably happens when it reaches 0.

For many values of gravity, the Y component of the linear velocity jumps from positive to negative without ever being very close to 0.

**Comment by stephengold** [comment]
`setSweepTest(false)` doesn't prevent the crash.

**Comment by stephengold** [comment]
A simple fix might be to replace the `normalize()` call in `walkStacklessQuantizedTreeAgainstRay()` with `safeNormalize()`.

**Comment by stephengold** [comment]
With that change, there's a similar crash at line 557:
```
#3  in __GI___assert_fail() at assert.c:101
#4  in btVector3::normalize()
      at /home/sgold/Git/Libbulletjme/src/main/native/bullet3/LinearMath/btVector3.h:305
#5  in btQuantizedBvh::walkStacklessQuantizedTreeAgainstRay() const
      at /home/sgold/Git/Libbulletjme/src/main/native/bullet3/BulletCollision/BroadphaseCollision/btQuantizedBvh.cpp:557
```

**Comment by stephengold** [comment]
With those 2 changes, the test no longer crashes.

**Comment by stephengold** [comment]
I've submitted PR 2634 to https://github.com/bulletphysics/bullet3 with the fix.

**Comment by stephengold** [comment]
Fixed in Minie at 25cfa032c6706aa354d3ab07f6605546d9da90cb
